src/main/java/chatsy/CommandHandler.java

```diff
@@ -0,0 +1,166 @@
+                ui.output(taskManager.listTasks());
+                break;
+            case "mark":
+                handleMarkUnmark(parts, true);
+                break;
+            case "unmark":
+                handleMarkUnmark(parts, false);
+                break;
+            case "delete":
+                handleDelete(parts);
+                break;
+            case "todo":
+                handleTodoTask(parts);
+                break;
+            case "deadline":
+                handleDeadlineTask(parts);
+                break;
+            case "event":
+                handleEventTask(parts);
+                break;
+            case "find":
+                handleFindTask(parts);
+                break;
+            default:
+                throw new InvalidCommandException();
+        }
+    }
+
+    private void handleMarkUnmark(String[] parts, boolean isMark) throws InvalidTaskIndexException {
+        if (parts.length > 1) {
+            int taskNumber = Integer.parseInt(parts[1]);
+            if (isMark) {
+                taskManager.markTask(taskNumber);
+                ui.output("Nice! I've marked this task as done:\n  " + taskManager.getTasks().get(taskNumber - 1));
+            } else {
+                taskManager.unmarkTask(taskNumber);
+                ui.output("OK, I've marked this task as not done yet:\n  " + taskManager.getTasks().get(taskNumber - 1));
+            }
+        } else {
+            ui.output("Please specify the task number to " + (isMark ? "mark." : "unmark."));
+        }
+    }
+
+    private void handleDelete(String[] parts) throws InvalidTaskIndexException {
+        if (parts.length > 1) {
+            int taskNumber = Integer.parseInt(parts[1]);
+            taskManager.deleteTask(taskNumber);
+            ui.output("Noted. I've removed this task.\nNow you have " + taskManager.getTasks().size() + " tasks in the list.");
+        } else {
+            ui.output("Please specify the task number to delete.");
+        }
+    }
+
+    private void handleTodoTask(String[] parts) throws EmptyDescriptionException {
+        if (parts.length > 1) {
+            taskManager.addTask(new TodoTask(parts[1]));
+            ui.output("Got it. I've added this task.\nNow you have " + taskManager.getTasks().size() + " tasks in the list.");
+        } else {
+            throw new EmptyDescriptionException();
+        }
+    }
+
+    private void handleDeadlineTask(String[] parts) throws EmptyDescriptionException {
+        if (parts.length > 1) {
+            String[] deadlineParts = parts[1].split(" /by ", 2);
+            if (deadlineParts.length > 1) {
+                try {
+                    LocalDate by = LocalDate.parse(deadlineParts[1], DATE_FORMATTER);
+                    taskManager.addTask(new DeadlineTask(deadlineParts[0], by));
+                    ui.output("Got it. I've added this task.\nNow you have " + taskManager.getTasks().size() + " tasks in the list.");
```

Comment:
Please try to use line wrapping at appropriate places of the line, as line length should be no longer than 120 chars.